orc8r/gateway/python/magma/ctraced/trace_manager.py

```diff
@@ -212,39 +309,29 @@ def _build_trace_filename(self):
             int(time.time()),
             _TRACE_FILE_EXT)
 
-    def _execute_start_trace_command(self, command: List[str]) -> bool:
-        """Executes a command to start a call trace
+    def _should_postprocess(self) -> bool:
+        return len(self._display_filters) > 0
 
-        Args:
-            command: Shell command with each token ordered in the list.
-              example: ["tshark", "-i", "eth0"] would be for "tshark -i eth0"
-
-        Returns:
-            True if successfully executed command
-        """
-        logging.info("TraceManager: Starting trace with %s, command: [%s]",
-                     self._tool_name, ' '.join(command))
-
-        self._ensure_trace_directory_exists()
-
-        # TODO(andreilee): Handle edge case where only one instance of the
-        #                  process can be running, and may have been started
-        #                  by something external as well.
-        try:
-            self._proc = subprocess.Popen(
-                command,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.STDOUT)
-        except SubprocessError as e:
-            logging.error("TraceManager: Failed to start trace: %s", str(e))
-            return False
+    def _dump_trace_logs(self):
+        logging.debug("TraceManager: Trace logs:")
+        logging.debug("<" * 25)
+        while True:
+            line = self._proc.stdout.readline()
+            if not line:
+                break
+            logging.debug("| %s", str(line.rstrip()))
+        logging.debug(">" * 25)
 
-        self._is_active = True
-        logging.info("TraceManager: Successfully started trace with %s",
-                     self._tool_name)
-        return True
+    def _wait_til_trace_file_exists(self):
+        while True:
+            if self._does_trace_file_exist():
+                logging.info("TraceManager: Trace file written!")
+                break
+            logging.info("TraceManager: Waiting 1s for trace file to be "
+                         "written...")
+            time.sleep(1)
 
-    def _ensure_trace_file_exists(self) -> bool:
+    def _does_trace_file_exist(self) -> bool:
         return os.path.isfile(self._trace_filename)
 
     def _ensure_tmp_file_deleted(self):
```

Comment:
In the below code if we hit exception when removing _trace_filename, we might leave _trace_filename_postprocessed around. Can you ensure we cleanup both the files. Thanks.